src/SME.SGP.Aplicacao/Comandos/ComandosAtribuicaoCJ.cs

```diff
@@ -10,26 +11,42 @@ public class ComandosAtribuicaoCJ : IComandosAtribuicaoCJ
     {
         private readonly IRepositorioAtribuicaoCJ repositorioAtribuicaoCJ;
         private readonly IServicoAtribuicaoCJ servicoAtribuicaoCJ;
+        private readonly IServicoEOL servicoEOL;
 
-        public ComandosAtribuicaoCJ(IRepositorioAtribuicaoCJ repositorioAtribuicaoCJ, IServicoAtribuicaoCJ servicoAtribuicaoCJ)
+        public ComandosAtribuicaoCJ(IRepositorioAtribuicaoCJ repositorioAtribuicaoCJ, IServicoAtribuicaoCJ servicoAtribuicaoCJ, IServicoEOL servicoEOL)
         {
             this.repositorioAtribuicaoCJ = repositorioAtribuicaoCJ ?? throw new ArgumentNullException(nameof(repositorioAtribuicaoCJ));
             this.servicoAtribuicaoCJ = servicoAtribuicaoCJ ?? throw new ArgumentNullException(nameof(servicoAtribuicaoCJ));
+            this.servicoEOL = servicoEOL ?? throw new ArgumentNullException(nameof(servicoEOL));
         }
 
         public async Task Salvar(AtribuicaoCJPersistenciaDto atribuicaoCJPersistenciaDto)
         {
             var atribuicoesAtuais = await repositorioAtribuicaoCJ.ObterPorFiltros(atribuicaoCJPersistenciaDto.Modalidade, atribuicaoCJPersistenciaDto.TurmaId,
                atribuicaoCJPersistenciaDto.UeId, 0, atribuicaoCJPersistenciaDto.UsuarioRf, string.Empty, null);
 
+            bool atribuiuCj = false;
+
             foreach (var atribuicaoDto in atribuicaoCJPersistenciaDto.Disciplinas)
             {
                 var atribuicao = TransformaDtoEmEntidade(atribuicaoCJPersistenciaDto, atribuicaoDto);
 
                 await servicoAtribuicaoCJ.Salvar(atribuicao, atribuicoesAtuais);
+
+                atribuiuCj = await AtribuirPerfilCJ(atribuicaoCJPersistenciaDto, atribuiuCj);
```

Comment:
Precisa ser dentro do for se é sempre para o mesmo usuário?

Reply:
Se o Salvar for realizado pelo menos 1x ele tem que atribuir o perfil CJ, independentemente se um dos itens subsequente falhar. por isso a flag atribuiuCj